daffodil-lib/src/main/scala/org/apache/daffodil/cookers/EntityReplacer.scala

```diff
@@ -616,18 +632,12 @@ class SingleCharacterLiteralNoCharClassEntitiesWithByteEntities(pn: String = nul
   extends SingleCharacterLiteralNoCharClassEntities(pn, allowByteEntities = true)
 
 class StringLiteralESEntityWithByteEntities(pn: String)
-  extends StringLiteralNoCharClassEntities(pn, allowByteEntities = true) {
+  extends StringLiteralNoCharClassEntities(pn, allowByteEntities = true)
+  with DisallowedCharClassEntitiesMixin {
 
-  /**
-   * Override Because ES is allowed.
-   */
-  override protected def noCharClassEntities(raw: String, context: ThrowsSDE): Unit = {
-    context.schemaDefinitionUnless(!raw.contains("%NL;"), "Property dfdl:%s cannot contain %%NL;", propName)
-    // ES is allowed
-    context.schemaDefinitionUnless(!raw.contains("%WSP;"), "Property dfdl:%s cannot contain %%WSP;", propName)
-    context.schemaDefinitionUnless(!raw.contains("%WSP+;"), "Property dfdl:%s cannot contain %%WSP+;", propName)
-    context.schemaDefinitionUnless(!raw.contains("%WSP*;"), "Property dfdl:%s cannot contain %%WSP*;", propName)
-  }
+  override val disallowedCharClassEntities = Seq("NL", "WSP", "WSP+", "WSP*")
+  override def testRaw(raw: String, context: ThrowsSDE) =
+    super[DisallowedCharClassEntitiesMixin].testRaw(raw, context)
 }
```

Comment:
We got code not covered by tests warnings for lines 638 and 640.  That means this class (StringLiteralESEntityWithByteEntities) isn't covered by any tests.  In fact, StringLiteralESEntityWithByteEntities isn't used anywhere in the codebase as far as I can see.  Should we remove it?

Reply:
Skimming through these, it actually looks like we have a few more these classes that aren't ever used.

We might want to just create a new cleanup ticket to go through all those and remove the ones that aren't used or figure out why some aren't used. It's possible we're just missing some cases where these were expected to be used and never implemented them.